src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/NativeWrapper.cs

```diff
@@ -482,7 +482,8 @@ internal static EventLogHandle EvtGetPublisherMetadataPropertyHandle(EventLogHan
         public static string EvtFormatMessage(EventLogHandle handle, uint msgId)
         {
             int bufferNeeded;
-            bool status = UnsafeNativeMethods.EvtFormatMessage(handle, EventLogHandle.Zero, msgId, 0, null, UnsafeNativeMethods.EvtFormatMessageFlags.EvtFormatMessageId, 0, null, out bufferNeeded);
+            char[] emptyBuffer = { '\0' };  // issue: https://github.com/dotnet/runtime/issues/100198
```

Comment:
This will allocate a new array on each call. Does this buffer get modified by EvtFormatMessage when the BufferSize is 0? If not, can it be a singleton array that's reused on every access?

Reply:
Yes, it's modified.  EvtFormatMessage will call memcopy to it, though in my debugging observations it's always copying zero (null character).  Because of this I was reluctant to use a singleton.  I also considered stackalloc, but felt it added a bit of complexity / unsafe code.  FWIW we previously allocated a StringBuilder here with a default array size of 16 characters.

Reply:
I was able to change to stackalloc, and I confirmed that changing the PInvoke to use `Span<char>` instead of `out char[]` didn't introduce any copy into the generated code.